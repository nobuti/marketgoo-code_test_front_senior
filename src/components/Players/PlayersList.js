import React from 'react'
import { Table, TableCell, TableRow, ButtonIcon, Icon } from '@marketgoo/ola'
import PropTypes from 'prop-types'

import Loading from './PlayersLoading'

const List = ({ data, isLoading = false, deleteAction = () => {} }) => {
  if (isLoading) {
    return <Loading />
  }

  const players = data ? data.sort((a, b) => b.score - a.score) : []

  return <Table
    responsive
  >
    <thead>
      <TableRow>
        <TableCell
          header
        >
            Player
        </TableCell>
        <TableCell
          header
        >
            Team
        </TableCell>
        <TableCell
          header
          variant="numeric"
        >
            Score
        </TableCell>
        <TableCell
          header
          variant="action"
        >
            Actions
        </TableCell>
      </TableRow>
    </thead>
    <tbody>
      { players.map(player =>
        <TableRow
          key={player.id}
          data-testid="row"
        >
          <TableCell
            variant="multiline"
          >
            {player.name}
          </TableCell>
          <TableCell
          >
            {player.team}
          </TableCell>
          <TableCell
            variant="numeric"
          >
            {player.score}
          </TableCell>
          <TableCell
            variant="action"
          >
            <ButtonIcon
              as="button"
              variant="primary"
              onClick={() => deleteAction(player.id)}
              data-testid="action"
            >
              <Icon
                name="close"
                size="small"
              />
            </ButtonIcon>
          </TableCell>
        </TableRow>
      )}
    </tbody>
  </Table>
}

List.displayName = 'PlayersList'

List.propTypes = {
  data: PropTypes.array.isRequired,
  isLoading: PropTypes.bool,
  deleteAction: PropTypes.func
}

export default List
