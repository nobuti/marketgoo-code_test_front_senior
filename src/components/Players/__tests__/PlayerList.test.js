import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import PlayersList from '../PlayersList'

describe('PlayersList', () => {
  const players = [
    {
      id: 1,
      name: 'Peter',
      team: 'Pandas',
      score: 16921,
      createdAt: '2020-07-08 02:25:52.976 +00:00',
      updatedAt: '2020-07-08 08:05:45.004 +00:00'
    },
    {
      id: 2,
      name: 'Jhon',
      team: 'Cobrakay',
      score: 15946,
      createdAt: '2020-07-08 02:25:52.977 +00:00',
      updatedAt: '2020-07-08 08:05:30.004 +00:00'
    },
    {
      id: 3,
      name: 'Tommy',
      team: 'Space',
      score: 16112,
      createdAt: '2020-07-08 02:25:52.977 +00:00',
      updatedAt: '2020-07-08 08:05:25.004 +00:00'
    }
  ]

  const deleteAction = jest.fn()

  test('renders properly', () => {
    const { container } = render(
      <PlayersList data={players} />
    )

    expect(container).toMatchSnapshot()
    expect(screen.queryAllByTestId('row')).toHaveLength(players.length)
  })

  test('renders loading properly', () => {
    const { container } = render(
      <PlayersList data={[]} isLoading={true} />
    )

    expect(container).toMatchSnapshot()
    expect(screen.queryAllByTestId('loading')).toHaveLength(1)
  })

  test('calls deleteAction properly', () => {
    render(
      <PlayersList data={players} deleteAction={deleteAction} />
    )

    const [firstButton] = screen.queryAllByTestId('action')
    fireEvent.click(firstButton)

    expect(deleteAction).toHaveBeenCalledWith(players[0].id)
  })
})
