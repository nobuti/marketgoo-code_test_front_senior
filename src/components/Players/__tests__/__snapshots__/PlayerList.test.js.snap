// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayersList renders loading properly 1`] = `
<div>
  <div
    class="Loading"
    data-testid="loading"
  >
    <svg
      class="ola_spinner"
      height="28"
      viewBox="0 0 28 28"
      width="28"
    >
      <circle
        cx="14"
        cy="14"
        r="12"
      />
    </svg>
    <span
      class="Loading-label"
      data-testid="loading-label"
    >
      Loading players
    </span>
  </div>
</div>
`;

exports[`PlayersList renders properly 1`] = `
<div>
  <div
    class="ola_table-container is-responsive"
  >
    <table
      class="ola_table"
    >
      <thead>
        <tr
          class="ola_tableRow"
        >
          <th
            class="ola_tableCell is-left"
          >
            Player
          </th>
          <th
            class="ola_tableCell is-left"
          >
            Team
          </th>
          <th
            class="ola_tableCell is-numeric"
          >
            Score
          </th>
          <th
            class="ola_tableCell is-action"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="ola_tableRow"
          data-testid="row"
        >
          <td
            class="ola_tableCell is-multiline"
          >
            Peter
          </td>
          <td
            class="ola_tableCell is-left"
          >
            Pandas
          </td>
          <td
            class="ola_tableCell is-numeric"
          >
            16921
          </td>
          <td
            class="ola_tableCell is-action"
          >
            <button
              class="ola_buttonIcon is-primary"
              data-testid="action"
            >
              <svg
                class="ola_icon is-small"
                fill-rule="evenodd"
                height="28"
                viewBox="0 0 28 28"
                width="28"
              >
                <path
                  d="M22.92 4c.296 0 .55.105.762.314.212.21.318.462.318.758s-.106.55-.318.762l-8.17 8.174 8.17 8.166c.212.212.318.464.318.754 0 .296-.106.549-.318.758-.212.21-.466.314-.762.314s-.547-.103-.753-.31l-8.171-8.166-8.163 8.166c-.206.207-.46.31-.761.31a1.04 1.04 0 0 1-.758-.31A1.02 1.02 0 0 1 4 22.936c0-.301.103-.555.31-.762l8.17-8.166-8.17-8.174A1.026 1.026 0 0 1 4 5.08c0-.296.105-.55.314-.762.21-.212.462-.318.758-.318.295 0 .55.106.761.318l8.163 8.174 8.17-8.174A1.03 1.03 0 0 1 22.92 4z"
                />
              </svg>
            </button>
          </td>
        </tr>
        <tr
          class="ola_tableRow"
          data-testid="row"
        >
          <td
            class="ola_tableCell is-multiline"
          >
            Jhon
          </td>
          <td
            class="ola_tableCell is-left"
          >
            Cobrakay
          </td>
          <td
            class="ola_tableCell is-numeric"
          >
            15946
          </td>
          <td
            class="ola_tableCell is-action"
          >
            <button
              class="ola_buttonIcon is-primary"
              data-testid="action"
            >
              <svg
                class="ola_icon is-small"
                fill-rule="evenodd"
                height="28"
                viewBox="0 0 28 28"
                width="28"
              >
                <path
                  d="M22.92 4c.296 0 .55.105.762.314.212.21.318.462.318.758s-.106.55-.318.762l-8.17 8.174 8.17 8.166c.212.212.318.464.318.754 0 .296-.106.549-.318.758-.212.21-.466.314-.762.314s-.547-.103-.753-.31l-8.171-8.166-8.163 8.166c-.206.207-.46.31-.761.31a1.04 1.04 0 0 1-.758-.31A1.02 1.02 0 0 1 4 22.936c0-.301.103-.555.31-.762l8.17-8.166-8.17-8.174A1.026 1.026 0 0 1 4 5.08c0-.296.105-.55.314-.762.21-.212.462-.318.758-.318.295 0 .55.106.761.318l8.163 8.174 8.17-8.174A1.03 1.03 0 0 1 22.92 4z"
                />
              </svg>
            </button>
          </td>
        </tr>
        <tr
          class="ola_tableRow"
          data-testid="row"
        >
          <td
            class="ola_tableCell is-multiline"
          >
            Tommy
          </td>
          <td
            class="ola_tableCell is-left"
          >
            Space
          </td>
          <td
            class="ola_tableCell is-numeric"
          >
            16112
          </td>
          <td
            class="ola_tableCell is-action"
          >
            <button
              class="ola_buttonIcon is-primary"
              data-testid="action"
            >
              <svg
                class="ola_icon is-small"
                fill-rule="evenodd"
                height="28"
                viewBox="0 0 28 28"
                width="28"
              >
                <path
                  d="M22.92 4c.296 0 .55.105.762.314.212.21.318.462.318.758s-.106.55-.318.762l-8.17 8.174 8.17 8.166c.212.212.318.464.318.754 0 .296-.106.549-.318.758-.212.21-.466.314-.762.314s-.547-.103-.753-.31l-8.171-8.166-8.163 8.166c-.206.207-.46.31-.761.31a1.04 1.04 0 0 1-.758-.31A1.02 1.02 0 0 1 4 22.936c0-.301.103-.555.31-.762l8.17-8.166-8.17-8.174A1.026 1.026 0 0 1 4 5.08c0-.296.105-.55.314-.762.21-.212.462-.318.758-.318.295 0 .55.106.761.318l8.163 8.174 8.17-8.174A1.03 1.03 0 0 1 22.92 4z"
                />
              </svg>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
